examples/nbody-runtime.rs

```diff
@@ -0,0 +1,290 @@
+                use stdsimd::arch::x86_64::*;
+                let t: f32x2 = (*s).into();
+
+                let u: f64x4 = unsafe {
+                    let res = _mm_rsqrt_ps(_mm_setr_ps(
+                        t.extract(0),
+                        t.extract(1),
+                        0.,
+                        0.,
+                    ));
+                    f32x4::from_bits(res).into()
+                };
+                f64x2::new(u.extract(0), u.extract(1))
+            }
+        }
+    } else {
+        impl Frsqrt for X86SSEFrsqrt {}
+    }
+}
+
+#[allow(dead_code)]
+struct ArmNeonFrsqrt {}
+
+cfg_if! {
+    if #[cfg(all(target_arch = "aarch64", target_feature = "neon"))] {
+        impl Frsqrt for ArmNeonFrsqrt {
+            fn frsqrt(&self, s: &f64x2) -> f64x2 {
+                #[cfg(target_arch = "aarch64")]
+                use stdsimd::arch::aarch64::*;
+                #[cfg(target_arch = "arm")]
+                use stdsimd::arch::arm::*;
+
+                let t: f32x2 = (*s).into();
+                let t: f32x2 = unsafe { vrsqrte_f32(t.into_bits()).into_bits() };
+                t.into()
+            }
+        }
+    } else {
+        impl Frsqrt for ArmNeonFrsqrt {}
+    }
+}
+
+cfg_if! {
+    if #[cfg(any(target_arch = "x86", target_arch = "x86_64"))] {
+        fn setup() -> Box<Frsqrt> {
+            if is_x86_feature_detected!("sse") {
+                Box::new(X86SSEFrsqrt {})
+            } else {
+                Box::new(FallbackFrsqrt {})
+            }
+        }
+    } else if #[cfg(target_arch = "arm")] {
+        fn setup() -> Box<Frsqrt> {
+            if is_arm_feature_detected!("neon") {
+                f = Box::new(ArmNeonFrsqrt {});
+            } else {
+                Box::new(FallbackFrsqrt {})
+            }
+        }
+    } else if #[cfg(target_arch = "aarch64")] {
+        fn setup() -> Box<Frsqrt> {
+            if is_aarch64_feature_detected!("neon") {
+                f = Box::new(ArmNeonFrsqrt {});
+            } else {
+                Box::new(FallbackFrsqrt {})
+            }
+        }
+    } else {
+        fn setup() -> Box<Frsqrt> {
+            Box::new(FallbackFrsqrt {})
+        }
+    }
+}
+
+struct Body {
+    x: [f64; 3],
+    _fill: f64,
+    v: [f64; 3],
+    mass: f64,
+}
+
+impl Body {
+    fn new(
+        x0: f64, x1: f64, x2: f64, v0: f64, v1: f64, v2: f64, mass: f64
+    ) -> Self {
+        Self {
+            x: [x0, x1, x2],
+            _fill: 0.0,
+            v: [v0, v1, v2],
+            mass,
+        }
+    }
+}
+
+const N_BODIES: usize = 5;
+const N: usize = N_BODIES * (N_BODIES - 1) / 2;
+fn offset_momentum(bodies: &mut [Body; N_BODIES]) {
+    let (sun, rest) = bodies.split_at_mut(1);
+    let sun = &mut sun[0];
+    for body in rest {
+        for k in 0..3 {
+            sun.v[k] -= body.v[k] * body.mass / SOLAR_MASS;
+        }
+    }
+}
+fn advance(f: &Box<Frsqrt>, bodies: &mut [Body; N_BODIES], dt: f64) {
```

Comment:
I'm not sure which is the least impacting way, we are going to use dynamic dispatch one way or another since that's what the runtime check requires in the end.

Reply:
Indeed dynamic dispatch is somewhere, but we don't want a dynamic dispatch per SIMD op but rather one per "large group of operations" which in this case is the entire program